x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/authc/ApiKeyIntegTests.java

```diff
@@ -274,6 +274,8 @@ public void testCreateApiKey() throws Exception {
         final long daysBetween = ChronoUnit.DAYS.between(start, expiration);
         assertThat(daysBetween, is(7L));
 
+        assertThat(getApiKeyDocument(response.getId()).get("type"), equalTo("rest"));
```

Comment:
I chose to avoid changes to `ApiKeyIntegTests` in this PR because:
1. The new API key type is not yet reflected in the Get/Query response so that some of the common verification code here won't work
2. The new API keys are not meant to be updated using existing APIs which are tested heavily here
3. The new type API keys are not meant to authenticate normally which is necessary for some of the tests here. 

I plan to take another look at this test class to see whether it can be factored into testing both type of API keys when all API changes are in place. For now I am keeping tests in other places which also makes them more explicit. Also, this test class is pretty terrible to work with (or review) ...